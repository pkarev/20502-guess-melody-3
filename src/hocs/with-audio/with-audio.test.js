import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withAudio from './with-audio.jsx';

const MockComponent = ({children}) => (
  <div>{children}</div>
);

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

const MockComponentWrapped = withAudio(MockComponent);

it(`Render WithAudio HOC`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped
          isPlaying={false}
          src={``}
          onPlayerClick={() => {}}
        />, {
          createNodeMock: () => ({})
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
