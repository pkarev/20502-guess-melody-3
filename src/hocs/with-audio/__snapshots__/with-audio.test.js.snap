// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render WithAudio HOC 1`] = `
<div>
  <audio />
</div>
`;
